Fix maven-source-plugin configuration for releases

`maven-source-plugin` v3.3.0 introduces a [new validation](https://issues.apache.org/jira/browse/MSOURCES-121) that highlighted a configuration error in the pom.
The error caused the sources to be added twice, this causes no issues because it's the same sources artifacts, but it's something that should be fixed.

```
    [INFO] <<< source:3.3.0:jar (attach-sources) < generate-sources @ asciidoctor-maven-plugin <<<
    [INFO] 
    [INFO] 
    [INFO] --- source:3.3.0:jar (attach-sources) @ asciidoctor-maven-plugin ---
    [INFO] Building jar: /home/asalgadr/temp/asciidoctor-maven-plugin/target/checkout/target/asciidoctor-maven-plugin-2.2.4-sources.jar
    [INFO] 
    [INFO] --- source:3.3.0:jar-no-fork (attach-sources) @ asciidoctor-maven-plugin ---
    [ERROR] We have duplicated artifacts attached.
    [INFO] ------------------------------------------------------------------------
    [INFO] BUILD FAILURE
    [INFO] ------------------------------------------------------------------------

```

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -25,6 +25,10 @@ Improvements::
   * Reimplement resource copy using 'plexus.util.DirectorScanner' instead of 'maven-filtering' to reduce dependencies and build time (#597)
   * Set minimal Maven version to v3.8.5 (#629)
 
+Build / Infrastructure::
+
+  * Fix maven-source-plugin configuration for releases (#644)
+
 Documentation::
 
   * Add v3.x.x migration guide (#598)

--- a/pom.xml
+++ b/pom.xml
@@ -228,7 +228,7 @@
                             <execution>
                                 <id>attach-sources</id>
                                 <goals>
-                                    <goal>jar</goal>
+                                    <goal>jar-no-fork</goal>
                                 </goals>
                             </execution>
                         </executions>

